app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/AacRecorder.kt

```diff
@@ -42,48 +22,31 @@ class AacRecorder(var voiceAnalysisOption: VoiceAnalysisOption) {
         )
         Timber.i("recording started")
         val buffer = ByteArray(bufferSize)
-        while (!shouldStopRecording) {
-            if (audioRecord?.recordingState == AudioRecord.RECORDSTATE_STOPPED) audioRecord?.startRecording()
-            audioRecord?.read(buffer, 0, buffer.size)
-            addData(buffer)
-        }
-    }
-
-    fun release() {
-        Timber.i("release recording")
-        shouldStopRecording = true
-        audioRecord?.stop()
-        audioRecord?.release()
-        audioRecord = null
-    }
-
-    @RunsInBackground
-    fun addData(array: ByteArray) {
-        newData = newData.plus(bytesToShorts(array).toTypedArray())
-        rawData = rawData.plus(array.toTypedArray())
 
-        if (newData.size >= NoiseDetector.DATA_SIZE &&
-            voiceAnalysisOption == VoiceAnalysisOption.NOISE_DETECTION
-        ) {
-            soundDetectionData.onNext(newData.takeLast(NoiseDetector.DATA_SIZE).toShortArray())
+        emitter.setCancellable {
+            shouldStopRecording = true
+            audioRecord.stop()
+            audioRecord.release()
+            Timber.i("release")
         }
 
-        if (newData.size >= MachineLearning.DATA_SIZE) {
-            machineLearningData.onNext(rawData.toByteArray() to newData.toShortArray())
-            clearRecordingData()
+        while (!shouldStopRecording) {
+            if (audioRecord.recordingState == AudioRecord.RECORDSTATE_STOPPED) audioRecord.startRecording()
+            audioRecord.read(buffer, 0, buffer.size)
+            emitter.onNext(buffer)
         }
     }
 
-    private fun clearRecordingData() {
-        newData = emptyArray()
-        rawData = emptyArray()
-    }
-
-    @RunsInBackground
-    private fun bytesToShorts(bytes: ByteArray): ShortArray {
-        val shorts = ShortArray(bytes.size / 2)
-        ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN).asShortBuffer().get(shorts)
-        return shorts
+    private fun findBestBufferSize(): Int {
+        var bufferSize = AudioRecord.getMinBufferSize(
+            SAMPLING_RATE,
+            AudioFormat.CHANNEL_IN_MONO,
+            AudioFormat.ENCODING_PCM_16BIT
+        ) * 2
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/92c7fea843a9bd05df3c694ba8c05996f0ca5e22/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/AacRecorder.kt#L45"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>